src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,124 @@
+                                + "date/time format and cannot be loaded into task list. ");
+                    }
+                    break;
+                case "E":
+                    String startString = tokens[3].trim();
+                    String endString = tokens[4].trim();
+                    try {
+                        DateTimeFormatter dateTimeFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+                        LocalDateTime start = LocalDateTime.parse(startString, dateTimeFormat);
+                        LocalDateTime end = LocalDateTime.parse(endString, dateTimeFormat);
+                        taskList.addTask(new Event(description, isCompleted, start, end));
+                    } catch (DateTimeParseException e) {
+                        System.out.println("Your event, \"" + description + "\" stored in file has incorrect "
+                                + "date/time format and cannot be loaded into task list. ");
+                    }
+                    break;
+                default:
+                    System.out.println("Invalid task type \"" + typeOfTask + "\" was found in file.");
+                }
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Uh oh! looks like the data in your file is corrupted and cannot be read! "
+                    + "Please verify data in the file to proceed. Proceeding without verification may cause "
+                    + "current data to disappear. \n");
+        }
+    }
+
+    /**
+     * Saves data from task list into file.
+     *
+     * @param taskList List that contains data to save into file
+     * @throws FileNotFoundException If file does not exist
+     */
+    public void saveFileData(TaskList taskList) throws FileNotFoundException {
+        File f = new File(this.filePath);
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(filePath));
+            for (int i = 0; i < taskList.size(); i++) {
+                Task task = taskList.get(i);
+                if (task instanceof Todo) {
+                    writer.write("T | " + (task.isCompleted() ? "1" : "0") + " | " + task.getDescription());
+                }
+                if (task instanceof Deadline) {
+                    writer.write("D | " + (task.isCompleted() ? "1" : "0") + " | " + task.getDescription()
+                            + " | " + ((Deadline) task).getByString());
+                }
+                if (task instanceof Event) {
+                    writer.write("E | " + (task.isCompleted() ? "1" : "0") + " | " + task.getDescription()
+                            + " | " + ((Event) task).getStartString() + " | " + ((Event) task).getEndString());
+                }
```

Comment:
I think you can include this method in the classes instead and use polymorphism from the Task class method

Reply:
Thank you for your advice :) I will try implementing that